File: app/src/main/java/com/example/breezil/favnews/db/AppDatabase.java
package com.example.breezil.favnews.db;


import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.example.breezil.favnews.model.Articles;
import com.example.breezil.favnews.model.BookMark;
import com.example.breezil.favnews.model.Source;

@Database(entities = {BookMark.class}, version = 1)
public abstract class AppDatabase  extends RoomDatabase {
    private static  AppDatabase appDatabase;
    public abstract BookMarkDao bookMarkDao();

    public static synchronized AppDatabase getAppDatabase(Context context){
        if(appDatabase == null){
            appDatabase = Room.databaseBuilder(context.getApplicationContext(),
                    AppDatabase.class, "article_db")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return appDatabase;
    }
}

File: app/src/main/java/com/example/breezil/fave/db/AppDatabase.kt
package com.example.breezil.fave.db


import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context

import com.example.breezil.fave.model.Articles
import com.example.breezil.fave.model.BookMark

@Database(entities = [BookMark::class, Articles::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun bookMarkDao(): BookMarkDao
    abstract fun articleDao(): ArticleDao

    companion object {
        private var appDatabase: AppDatabase? = null

        @Synchronized
        fun getAppDatabase(context: Context): AppDatabase {
            if (appDatabase == null) {
                appDatabase = Room.databaseBuilder(context.applicationContext,
                        AppDatabase::class.java, "article_db")
                        .fallbackToDestructiveMigration()
                        .build()
            }
            return appDatabase as AppDatabase
        }
    }
}
